### ISSUE TITLE
InterstitialAd onAdDismissedFullScreenContent not working on Android

### ISSUE THREAD

**Issue by Jaew00Shin** [googleads/googleads-mobile-flutter#1213]
### Step 0: Are you in the right place?

  * For general technical questions, or help with project-specific issues like setting up ads in
   your app, reach out to our support team on the
   [Developer Forum](https://groups.google.com/forum/#!categories/google-admob-ads-sdk).
  * For assistance with your AdMob account, reach out to
   [AdMob Support](https://support.google.com/admob/?hl=en#topic=7383088).
  * For feedback on [our documentation](https://developers.google.com/admob/flutter/quick-start),
   send your feedback by pressing the **Send Feedback** button at the top right of the
   documentation page you are on.
  * For issues related to __the code in this repository__, continue filing this GitHub issue.

  * Once you've read this section and determined that your issue is appropriate for
    this repository, **please delete this section**.

---

## Plugin Version

google_mobile_ads: ^5.1.0

## Steps to Reproduce

```
Future<void> _loadInterstitialAd(String adUnitId) async {
    await AdManagerInterstitialAd.load(
      adUnitId: adUnitId,
      request: const AdManagerAdRequest(),
      adLoadCallback: AdManagerInterstitialAdLoadCallback(
        onAdLoaded: (ad) {
          _interstitialAds[adUnitId] = ad;
        },
        onAdFailedToLoad: (error) {},
      ),
    );
  }

  Future<void> showInterstitialAd({
    required String adUnitId,
    void Function(Ad ad)? onAdDismissedFullScreenContent,
    void Function(Ad ad, AdError error)? onAdFailedToShowFullScreenContent,
    void Function(Ad ad)? onAdShowedFullScreenContent,
    void Function(Ad ad)? onAdClicked,
  }) async {
    final interstitialAd = _interstitialAds[adUnitId];

    interstitialAd?.fullScreenContentCallback = FullScreenContentCallback(
      onAdDismissedFullScreenContent: (ad) async {
        onAdDismissedFullScreenContent?.call(ad);
        await ad.dispose();
        await _loadInterstitialAd(adUnitId);
      },
      onAdFailedToShowFullScreenContent: (ad, error) async {
        onAdFailedToShowFullScreenContent?.call(ad, error);
        await ad.dispose();
        await _loadInterstitialAd(adUnitId);
      },
      onAdShowedFullScreenContent: onAdShowedFullScreenContent,
      onAdClicked: onAdClicked,
    );

    await interstitialAd?.show();
  }
```

`onAdDismissedFullScreenContent` sometimes doesn't work.
It should be always called after calling `onAdShowedFullScreenContent` and after ad is dismissed.
But it sometimes works and sometimes doesn't work.

**Expected results:** 

It should be always called after calling `onAdShowedFullScreenContent` and after ad is dismissed.

**Actual results:** 

But it sometimes works and sometimes doesn't work.

<details>
  <summary>Logs</summary>

```
```

```
 flutter analyze
Analyzing rock2you...                                                   
No issues found! (ran in 17.5s)
```

```

[✓] Xcode - develop for iOS and macOS (Xcode 16.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16B40
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] IntelliJ IDEA Ultimate Edition (version 2022.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.93.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.96.0

[✓] Connected device (5 available)
    • SM G965N (mobile)               • 22385a1f0e037ece          • android-arm64  • Android 10 (API 29)
    • 신재우의 iPad (mobile)              • 00008103-000671983ED9001E • ios            • iOS 18.1.1 22B91
    • macOS (desktop)                 • macos                     • darwin-arm64   • macOS 15.1.1 24B91 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad     • darwin         • macOS 15.1.1 24B91 darwin-arm64
    • Chrome (web)                    • chrome                    • web-javascript • Google Chrome 131.0.6778.86
    ! Error: Browsing on the local area network for 신재우의 iPhone. Ensure the device is unlocked and attached with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)

[✓] Network resources
    • All expected network resources are available.

• No issues found!

```

</details>

**Comment by malandr2** [comment]
Hi @Jaew00Shin, can you replicate this behavior you're experiencing using our [interstitial_sample](https://github.com/googleads/googleads-mobile-flutter/tree/main/samples/admob/interstitial_example)? I am unable to

**Comment by Jaew00Shin** [comment]
I forgot to write an essential condition. It always works well on IOS. It sometimes doesn't work on Android.

**Comment by malandr2** [comment]
Hi @Jaew00Shin, I'm unable to replicate your use-case using the [interstitial_sample](https://github.com/googleads/googleads-mobile-flutter/tree/main/samples/admob/interstitial_example). It appears your code is a little different than our guide https://developers.google.com/admob/flutter/interstitial - are you able to replicate following the code in the guide or with our developer sample? Or can you provide a [minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)?

**Comment by Jaew00Shin** [comment]
```
 Future<void> _loadInterstitialAd(String adUnitId) async {
    await InterstitialAd.load(
      adUnitId: adUnitId,
      request: const AdManagerAdRequest(),
      adLoadCallback: InterstitialAdLoadCallback(
        onAdLoaded: (ad) {
          _interstitialAds[adUnitId] = ad;
        },
        onAdFailedToLoad: (error) {},
      ),
    );
  }

  Future<void> showInterstitialAd({
    required String adUnitId,
    void Function(Ad ad)? onAdDismissedFullScreenContent,
    void Function(Ad ad, AdError error)? onAdFailedToShowFullScreenContent,
    void Function(Ad ad)? onAdShowedFullScreenContent,
    void Function(Ad ad)? onAdClicked,
  }) async {
    final interstitialAd = _interstitialAds[adUnitId];

    interstitialAd?.fullScreenContentCallback = FullScreenContentCallback(
      onAdDismissedFullScreenContent: (ad) async {
        onAdDismissedFullScreenContent?.call(ad);
        await ad.dispose();
        await _loadInterstitialAd(adUnitId);
      },
      onAdFailedToShowFullScreenContent: (ad, error) async {
        onAdFailedToShowFullScreenContent?.call(ad, error);
        await ad.dispose();
        await _loadInterstitialAd(adUnitId);
      },
      onAdShowedFullScreenContent: onAdShowedFullScreenContent,
      onAdClicked: onAdClicked,
    );

    await interstitialAd?.show();
  }
```

Thanks for fixing me. I confused `InterstitialAd` as `AdManagerInterstitialAd`.
I fixed `AdManagerInterstitialAd` to `InterstitialAd`.
But the problem is not fixed at all. It sometimes works well but sometimes doesn't work.
I gave you the sample code. and I hope you recognize this issue.

**Comment by malandr2** [comment]
Hi @Jaew00Shin , I'm unable to reproduce - have you been able to reproduce in the [interstitial_sample](https://github.com/googleads/googleads-mobile-flutter/tree/main/samples/admob/interstitial_example)?

**Comment by malandr2** [comment]
Hi @Jaew00Shin, I copied your code into the [interstitial_sample](https://github.com/googleads/googleads-mobile-flutter/tree/main/samples/admob/interstitial_example) and was also able to see `onAdDismissedFullScreenContent` called every time. The only difference I changed was not using a dictionary to store the interstitial ads. I created an `_interstitialAd` variable to store the reference one at a time.
